src/legacy/core_plugins/data/public/index_patterns/index_patterns_service.ts

```diff
@@ -50,9 +54,9 @@ const savedObjectsClient = chrome.getSavedObjectsClient();
  * @internal
  */
 export class IndexPatternsService {
```

Comment:
@lukeelmers  do we have the the base (parent) class for 'Services'?

Reply:
You mean a class with lifecycle methods for each service?

Yes, we've been doing that by convention as it mirrors how Core code is structured, however this is simply convention and not actually enforced by the new platform.